src/dappkit/DappKitSignTxScreen.tsx

```diff
@@ -61,55 +68,57 @@ const DappKitSignTxScreen = ({ route }: Props) => {
     }
   }
 
-  const requestDetails = [
-    {
-      label: t('action.operation'),
-      value: t('transaction.signTX'),
-    },
-  ]
-
   return (
-    <View style={styles.container}>
+    <SafeAreaView edges={['bottom']} style={styles.container}>
       <RequestContent
         onAccept={handleAllow}
         onDeny={handleCancel}
         dappName={dappName}
         dappImageUrl={dappConnectInfo === DappConnectInfo.Basic ? activeDapp?.iconUrl : undefined}
-        title={t('confirmTransaction', { dappName })}
-        description={t('action.askingV1_35', { dappName })}
+        title={t('confirmTransaction')}
+        description={t('walletConnectRequest.signTransaction', { dappName })}
         testId="DappKitSignRequest"
         dappUrl={callback}
-        requestDetails={requestDetails}
       >
-        <Touchable testID="ShowTransactionDetailsButton" onPress={handleShowTransactionDetails}>
-          <Expandable isExpandable isExpanded={showTransactionDetails}>
-            <Text style={[styles.bodyText, styles.underLine]}>{t('action.details')}</Text>
-          </Expandable>
-        </Touchable>
-
-        {showTransactionDetails && (
-          <Text testID="DappData" style={styles.bodyText}>
-            {txs[0].txData}
-          </Text>
-        )}
+        <View style={styles.transactionContainer}>
+          <View style={styles.transactionDataContainer}>
+            <Text style={styles.transactionDataLabel}>
+              {t('walletConnectRequest.transactionDataLabel')}
+            </Text>
+            <Text testID="DappData" style={fontStyles.small} numberOfLines={1} ellipsizeMode="tail">
+              {txs[0].txData}
+            </Text>
+          </View>
+          <Touchable hitSlop={variables.iconHitslop} onPress={handleCopyRequestPayload}>
+            <CopyIcon />
+          </Touchable>
+        </View>
       </RequestContent>
-    </View>
+    </SafeAreaView>
   )
 }
 
 const styles = StyleSheet.create({
   container: {
     padding: Spacing.Thick24,
+    paddingBottom: 0, // SafeAreaView already adds enough space here
```

Comment:
same question here.

Reply:
yeah good point - this PR has some quick hacks (this is one of them) because i had the dynamic heights on the bottom sheets but took that out to get reviewed separately. the 0 padding bottom is removed in the follow up https://github.com/valora-inc/wallet/pull/3412 so if you're okay with it i'll not spend time changing this here

Reply:
Sounds good!